groovy-language-server/src/main/java/com/palantir/groovylanguageserver/GroovycWrapper.java

```diff
@@ -213,13 +301,46 @@ private static SymbolKind getKind(ClassNode node) {
         return SymbolKind.Class;
     }
 
-    private Set<SymbolInformation> getMethodSymbolInformations(String sourcePath, ClassNode parent, MethodNode method) {
+    private SymbolInformation getVariableSymbolInformation(String parentName, String sourcePath, Variable variable) {
+        Location location;
+        SymbolKind kind;
+        if (variable instanceof DynamicVariable) {
+            kind = SymbolKind.Field;
+            location = new LocationBuilder().uri(sourcePath).range(Ranges.UNDEFINED_RANGE).build();
+        } else if (variable instanceof FieldNode) {
+            kind = SymbolKind.Field;
+            location = createLocation(sourcePath, (FieldNode) variable);
+        } else if (variable instanceof Parameter) {
+            kind = SymbolKind.Variable;
+            location = createLocation(sourcePath, (Parameter) variable);
+        } else if (variable instanceof PropertyNode) {
+            kind = SymbolKind.Field;
+            location = createLocation(sourcePath, (PropertyNode) variable);
+        } else if (variable instanceof VariableExpression) {
+            kind = SymbolKind.Variable;
+            location = createLocation(sourcePath, (VariableExpression) variable);
+        } else {
+            throw new IllegalArgumentException(String.format("Unknown type of variable: %s", variable));
+        }
+        return createSymbolInformation(variable.getName(), kind, location,
+                Optional.of(parentName));
+    }
+
+    private Set<SymbolInformation> getMethodSymbolInformations(Map<String, Set<SymbolInformation>> newReferences,
+            String sourcePath, ClassNode parent, MethodNode method) {
         Set<SymbolInformation> symbols = Sets.newHashSet();
-        symbols.add(createSymbolInformation(method.getName(), SymbolKind.Method, createLocation(sourcePath, method),
-                Optional.of(parent.getName())));
+
+        SymbolInformation methodSymbol =
+                createSymbolInformation(method.getName(), SymbolKind.Method, createLocation(sourcePath, method),
+                        Optional.of(parent.getName()));
+        symbols.add(methodSymbol);
+        newReferences.computeIfAbsent(method.getReturnType().getName(), (value) -> Sets.newHashSet()).add(methodSymbol);
```

Comment:
maybe cleaner to add a small helper function called `addToValueSet` that takes in the map, the key and new value in. So that it's easier to know what the command is doing.